### PR TITLE
PWN-6052 - make a fragment with new search func

### PR DESCRIPTION
**Pull request by kabramovp2p**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-6052

## Description of Work

Make new search fragment implementation

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/send/ui/search/NewSearchFragment.kt [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+        showEmptyState(isEmpty = result.isEmpty())
+    }
+
+    override fun showMessage(textRes: Int?) = with(binding.messageTextView) {
+        isVisible = if (textRes == null) {
+            false
+        } else {
+            setText(textRes)
+            true
+        }
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/send/ui/search/NewSearchFragment.kt:157]: `binding.messageTextView withTextOrGone textRes`

Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/send/ui/search/NewSearchFragment.kt:157]: ohh you right! thx!

**Review by davran1001 — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/send/ui/search/NewSearchFragment.kt [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    private val searchAdapter: SearchAdapter by unsafeLazy {
+        SearchAdapter(
+            onItemClicked = presenter::onSearchResultClick,
+            usernameDomainFeatureToggle = get()
+        )
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        analyticsInteractor.logScreenOpenEvent(ScreenNames.Send.RECIPIENT_ADDRESS)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/send/ui/search/NewSearchFragment.kt:61]: better to add this log in BaseFragment

File: app/src/main/java/org/p2p/wallet/send/ui/search/NewSearchPresenter.kt [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        )
+
+        searchJob?.cancel()
+        searchJob = launch {
+            try {
+                delay(DELAY_IN_MS)
+                view?.showLoading(isLoading = true)
+                validateAndSearch(target)
+            } catch (e: CancellationException) {
+                Timber.w("Cancelled search target validation: ${target.value}")
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/send/ui/search/NewSearchPresenter.kt:65]: Timber.i

File: app/src/main/java/org/p2p/wallet/send/ui/search/NewSearchPresenter.kt [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+            } catch (e: Throwable) {
+                Timber.e(e, "Error searching target")
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/send/ui/search/NewSearchPresenter.kt:67]: Error searching target: $newQuery

File: app/src/main/java/org/p2p/wallet/send/ui/search/NewSearchPresenter.kt [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+    }
+
+    private suspend fun searchByUsername(username: String) {
+        val usernames = searchInteractor.searchByName(username)
+        if (usernames.isEmpty()) {
+            showNotFound()
+            return
+        }
+
+        view?.showMessage(R.string.search_found_header)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/send/ui/search/NewSearchPresenter.kt:112]: maybe `search_username_found`?

File: app/src/main/java/org/p2p/wallet/send/ui/search/NewSearchFragment.kt [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+import org.p2p.wallet.utils.withArgs
+
+class NewSearchFragment :
+    BaseMvpFragment<NewSearchContract.View, NewSearchContract.Presenter>(R.layout.fragment_new_search),
+    NewSearchContract.View {
+
+    companion object {
+        private const val EXTRA_USERNAMES = "EXTRA_USERNAMES"
+
+        fun create(usernames: List<SearchResult>? = null): NewSearchFragment =
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/send/ui/search/NewSearchFragment.kt:39]: always null, no?

Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/send/ui/search/NewSearchFragment.kt:39]: if it's for future 'popular recipients' there is not need for this arg either. 
you can fetch those from the repository

**Review by gslevinkov — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/p2p/wallet/send/ui/search/NewSearchFragment.kt (hunk 6 above)
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/send/ui/search/NewSearchFragment.kt:39]: it's when you go from send fragment and you already have some selected recipients

if this won't be needed we will remove it

File: app/src/main/java/org/p2p/wallet/common/feature_toggles/toggles/remote/NewSendEnabledFeatureToggle.kt [hunk 7]
```diff
@@ -0,0 +1,11 @@
+package org.p2p.wallet.common.feature_toggles.toggles.remote
+
+import org.p2p.wallet.common.feature_toggles.remote_config.RemoteConfigValuesProvider
+
+class NewSendEnabledFeatureToggle(
+    valuesProvider: RemoteConfigValuesProvider
+) : BooleanFeatureToggle(valuesProvider) {
+    override val featureKey: String = "keyapp_new_send_enabled"
+    override val featureDescription: String = "Enable New Send feature"
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/common/feature_toggles/toggles/remote/NewSendEnabledFeatureToggle.kt:9]: Enable new send flow maybe?

**Review by gslevinkov — APPROVED** [review]

**Pull request merged**